core/server/master/src/main/java/alluxio/master/block/BlockMasterWorkerServiceHandler.java

```diff
@@ -84,10 +84,14 @@ public void blockHeartbeat(BlockHeartbeatPRequest request,
                     e -> Block.BlockLocation.newBuilder().setTier(e.getKey().getTierAlias())
                         .setMediumType(e.getKey().getMediumType()).build(),
                     e -> e.getValue().getBlockIdList(),
+                    /**
+                     * The merger function is invoked on key collisions to merge the values.
+                     * In fact this merger should never be invoked because the list is deduplicated
+                     * by {@link BlockMasterClient#heartbeat} before sending to the master.
+                     */
                     (e1, e2) -> {
-                      List<Long> e3 = new ArrayList<>(e1);
-                      e3.addAll(e2);
-                      return e3;
+                      e1.addAll(e2);
+                      return e1;
```

Comment:
Although this is currently not invoked at all, I think the small improvement is needed to guard future changes. There's no need to create a new list and copy, we can just use the existing list.